<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>880</width>
    <height>655</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Kernighan-Lin Algoruthm Demo</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_9">
    <item row="0" column="0" rowspan="2">
     <widget class="QTabWidget" name="tabWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>220</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>220</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Auto</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Vertices</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QSpinBox" name="spinBox"/>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Edges</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QSpinBox" name="spinBox_3"/>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>Diameter</string>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QSpinBox" name="spinBox_4"/>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>TextLabel</string>
          </property>
         </widget>
        </item>
        <item row="3" column="2">
         <widget class="QSpinBox" name="spinBox_5"/>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>TextLabel</string>
          </property>
         </widget>
        </item>
        <item row="4" column="2">
         <widget class="QSpinBox" name="spinBox_6"/>
        </item>
        <item row="5" column="1">
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>73</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="6" column="0">
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>54</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="6" column="1">
         <widget class="QPushButton" name="pushButton_2">
          <property name="text">
           <string>Generate</string>
          </property>
         </widget>
        </item>
        <item row="6" column="2">
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>46</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
       <zorder>pushButton_2</zorder>
       <zorder>spinBox</zorder>
       <zorder>label_4</zorder>
       <zorder>spinBox_3</zorder>
       <zorder>spinBox_4</zorder>
       <zorder>spinBox_5</zorder>
       <zorder>spinBox_6</zorder>
       <zorder>label_5</zorder>
       <zorder>label_6</zorder>
       <zorder>label_7</zorder>
       <zorder>label_8</zorder>
       <zorder>horizontalSpacer</zorder>
       <zorder>horizontalSpacer_2</zorder>
       <zorder>verticalSpacer_3</zorder>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Manual</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_7">
        <item row="0" column="0">
         <widget class="QGroupBox" name="groupBox_4">
          <property name="title">
           <string>Vertex</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_6">
           <item row="0" column="0">
            <widget class="QLabel" name="label_22">
             <property name="text">
              <string>Name (id)</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="vertexId"/>
           </item>
           <item row="1" column="0" colspan="2">
            <widget class="QPushButton" name="addVertexButton">
             <property name="text">
              <string>Add</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Edge</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="0" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Vertex 1</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QComboBox" name="edgeVertex2Id"/>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Weight</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="edgeVertex1Id"/>
           </item>
           <item row="2" column="1">
            <widget class="QSpinBox" name="edgeWeight"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>Vertex 2</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0" colspan="2">
            <widget class="QPushButton" name="addEdgeButton">
             <property name="text">
              <string>Add</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="0" column="1" rowspan="4" colspan="2">
     <widget class="QGraphicsView" name="graphicsView"/>
    </item>
    <item row="0" column="3">
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="0">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>value1:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_12">
        <property name="text">
         <string>value2:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="label_13">
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_14">
        <property name="text">
         <string>value3:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="label_15">
        <property name="text">
         <string>0</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="3" rowspan="2">
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>219</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="2" column="0">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>Parallelism</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_8">
       <item row="0" column="0" colspan="2">
        <widget class="QLabel" name="label_19">
         <property name="text">
          <string>Cores on your computer:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QLabel" name="label_20">
         <property name="text">
          <string>4</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_21">
         <property name="text">
          <string>Threads:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1" colspan="2">
        <widget class="QSpinBox" name="spinBox_8"/>
       </item>
       <item row="2" column="0" colspan="3">
        <widget class="QSlider" name="horizontalSlider_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="3" column="0">
     <widget class="QGroupBox" name="groupBox_3">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Maximum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>220</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>220</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>Partitioning</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="0" column="0">
        <widget class="QLabel" name="label_17">
         <property name="text">
          <string>Number of classes:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <spacer name="horizontalSpacer_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="2">
        <widget class="QSpinBox" name="spinBox_7"/>
       </item>
       <item row="1" column="0" colspan="3">
        <widget class="QSlider" name="horizontalSlider_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QPushButton" name="pushButton_3">
         <property name="text">
          <string>Run</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1" colspan="2">
        <widget class="QPushButton" name="pushButton_4">
         <property name="text">
          <string>Reset</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="3" column="3" rowspan="2">
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QLabel" name="label_16">
        <property name="text">
         <string>Time sequential:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLCDNumber" name="lcdNumber">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>50</height>
         </size>
        </property>
        <property name="smallDecimalPoint">
         <bool>false</bool>
        </property>
        <property name="digitCount">
         <number>5</number>
        </property>
        <property name="mode">
         <enum>QLCDNumber::Dec</enum>
        </property>
        <property name="segmentStyle">
         <enum>QLCDNumber::Filled</enum>
        </property>
        <property name="intValue" stdset="0">
         <number>3876</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_18">
        <property name="text">
         <string>Time larallel:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLCDNumber" name="lcdNumber_2">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>50</height>
         </size>
        </property>
        <property name="smallDecimalPoint">
         <bool>false</bool>
        </property>
        <property name="digitCount">
         <number>5</number>
        </property>
        <property name="mode">
         <enum>QLCDNumber::Dec</enum>
        </property>
        <property name="segmentStyle">
         <enum>QLCDNumber::Filled</enum>
        </property>
        <property name="intValue" stdset="0">
         <number>3876</number>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="4" column="0">
     <spacer name="verticalSpacer_2">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>31</height>
       </size>
      </property>
     </spacer>
    </item>
    <item row="4" column="1">
     <widget class="QLabel" name="label_9">
      <property name="text">
       <string>Scale</string>
      </property>
     </widget>
    </item>
    <item row="4" column="2">
     <widget class="QSlider" name="horizontalSlider">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
   </layout>
   <zorder>graphicsView</zorder>
   <zorder>tabWidget</zorder>
   <zorder>horizontalSlider</zorder>
   <zorder>label_9</zorder>
   <zorder></zorder>
   <zorder>verticalSpacer</zorder>
   <zorder>verticalSpacer_2</zorder>
   <zorder>groupBox_3</zorder>
   <zorder>groupBox_2</zorder>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>880</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
